pkgs/development/r-modules/default.nix

```diff
@@ -1803,10 +1803,33 @@ let
       '';
     });
 
-    torch = old.torch.overrideAttrs (attrs: {
-      preConfigure = ''
-        patchShebangs configure
-      '';
+    torch = let
+      # Sets the correct string to download either the binary for
+      # the cpu version of the torch package, or the
+      # the cuda-enabled version.
+      # To use gpu acceleration, set `config.cudaSupport = true;`
+      # when importing nixpkgs in your shell
+       accel = if pkgs.config.cudaSupport then "cu118" else "cpu";
+
+       binary_sha = if pkgs.config.cudaSupport then
+         "sha256-a80sG89C0svZzkjNRpY0rTR2P1JdvKAbWDGIIghsv2Y=" else
+         "sha256-qUn8Rot6ME7iTvtNd52iw3ebqMnpLz7kwl/9GoPHD+I=";
+      in
+     old.torch.overrideAttrs (attrs: {
+       nativeBuildInputs = attrs.nativeBuildInputs or [ ] ++ [
+         pkgs.autoPatchelfHook
+         pkgs.autoAddDriverRunpath
+       ];
+      buildInputs = attrs.buildInputs or [ ] ++ [
+        "${lib.getLib R}/lib/R" # libR
+        pkgs.zlib
+      ] ++ lib.optionals pkgs.config.cudaSupport [
+        pkgs.cudaPackages.cuda_cudart
+      ];
```

Comment:
These changes are sufficient to do something like

```
> library(torch)
> torch_sum(torch_randn(c(10, 10), device="cuda"))
```

However, normally libtorch_cuda links e.g. libcublas, etc, which I didn't have to explicitly mention. I suppose that means they're not listed in DT_NEEDED and are loaded by dlopen instead. Either way I expect all related functionality to be broken and I can't allocate the time to do more tests unfortunately

I also see there's a lot of garbage in the outputs, e.g. a vendored copy of `libcudart.so`. You could link those by telling autoPatchelf to add `$ORIGIN`, but I'd advise against it and recommend that you delete the vendored libraries. Cuda libraries tend to need ad hoc patchelfing as well, we've had issues with that before. Cuda libraries that originate from PyPi are usually broken, which is why we link the ones from `cudaPackages` instead